main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java

```diff
@@ -48,27 +48,40 @@ public class StopTimeIncreasingDistanceValidator extends FileValidator {
   public void validate(NoticeContainer noticeContainer) {
     for (List<GtfsStopTime> stopTimeList : Multimaps.asMap(stopTimeTable.byTripIdMap()).values()) {
       // GtfsStopTime objects are sorted based on @SequenceKey annotation on stop_sequence field.
-      for (int i = 1; i < stopTimeList.size(); ++i) {
-        GtfsStopTime prev = stopTimeList.get(i - 1);
-        GtfsStopTime curr = stopTimeList.get(i);
-        if (prev.hasShapeDistTraveled()
+      // The last good is the last stopTime entity that has a stopId
+      GtfsStopTime lastGood = null;
+      for (var curr : stopTimeList) {
+        if (!curr.hasStopId()) {
+          continue;
+        }
+        if (lastGood == null) {
+          lastGood = curr;
+          continue;
+        }
```

Comment:
I thought only rows without stop IDs would be ignored. In this implementation, we are potentially increasing the travel distance by not considering the intermediate stops without stop IDs. Is this the expected results? cc: @tzujenchanmbd

Reply:
> I thought only rows without stop IDs would be ignored. I In this implementation, we are potentially increasing the travel distance by not considering the intermediate stops without stop IDs.

I probably don't understand the question here well.

`decreasing_or_equal_stop_time_distance` notice should only compare `shape_dist_traveled` value for rows with `stop_id`.  (**ignore** rows without `stop_id`)

And there should be another notice (this one) to be triggered if `shape_dist_traveled` value exists (no matter what value it is) in the row AND the row without `stop_id`.

Reply:
Reverted changes to StopTimeIncreasingDistanceValidator as it's covered in https://github.com/MobilityData/gtfs-validator/pull/1895